smartsim/experiment.py

```diff
@@ -923,6 +926,32 @@ def _launch_summary(self, manifest: Manifest) -> None:
 
         logger.info(summary)
 
+    def _create_dir_and_set_path(self, start_manifest: Manifest) -> None:
+        def create_dir_and_set_path(
+            entity: t.Union[Orchestrator, Model, Ensemble], entity_path: str
+        ) -> None:
+            if not os.path.isdir(entity_path):
+                os.makedirs(entity_path)
+                entity.set_path(entity_path)
+
+        for model in start_manifest.models:
+            create_dir_and_set_path(model, model.path)
+
+        for orch in start_manifest.dbs:
+            orch_path = osp.join(self.exp_path, orch.name)
+            create_dir_and_set_path(orch, orch_path)
+
+        for ensemble in start_manifest.ensembles:
+            ensemble_path = osp.join(self.exp_path, ensemble.name)
+            create_dir_and_set_path(ensemble, ensemble_path)
+
+            for member in ensemble.models:
+                # if member.path is None:
+                member_path = osp.join(self.exp_path, ensemble.name, member.name)
+                # else:
+                #     member_path = member.path
+                create_dir_and_set_path(member, member_path)
+
```

Comment:
Changing this function name to `create_entity_dir` and removing all `set_path` since paths are set in factory methods. The portion of the code that creates directories is in the `Generator` class - if not called no directories are created. If I do not create the directories here the experiment will throw an error that the path does not exists since before the path was defaulted to CWD. The telemetry monitor does create a experiment dir if DNE. I would suggest that we keep the dir generation here (unless you have another idea!). Telemetry should still create an Experiment directory bc if I create an Orchestrator using the constructor, the path is defaulted to CWD, no Exp will be created. I also do not believe I can use the Generator class since this uses set_paths rn to overwrite path arguments and also attached all generator files. But I might be convince to use the Generator function once the bug is fixed?